sunpy/timeseries/metadata.py

```diff
@@ -90,13 +100,14 @@ def __init__(self, meta=None, timerange=None, colnames=None):
                 self.metadata = copy.copy(meta)
         else:
             # In the event no metadata dictionary is sent we default to something usable
-            if isinstance(timerange, TimeRange) and isinstance(colnames, list):
-                self.metadata.append((timerange, colnames, MetaDict()))
-            elif isinstance(timerange, TimeRange):
-                self.metadata.append((timerange, [], MetaDict()))
-                warnings.warn("No time range given for metadata. "
-                              "This will mean the metadata can't be linked "
-                              "to columns in data.", SunpyUserWarning)
+            if isinstance(timerange, TimeRange):
+                if isinstance(colnames, list):
+                    self.metadata.append((timerange, colnames, MetaDict()))
+                else:
+                    self.metadata.append((timerange, [], MetaDict()))
+                    warnings.warn("No time range given for metadata. "
+                                  "This will mean the metadata can't be linked "
+                                  "to columns in data.", SunpyUserWarning)
```

Comment:
columns not rows? I am a little confused by this.

Reply:
🤷‍♂ this is just what was there to begin with, I haven't changed it.